### PR TITLE
Migrate whole README to the documentation

### PR DESCRIPTION
**Pull request by almogtavor**

### CODE REVIEW

File: docs/src/.vuepress/config.js [hunk 1]
```diff
@@ -35,6 +35,8 @@ module.exports = {
     docsDir: "",
     editLinkText: "",
     lastUpdated: false,
+    logo:
+      "https://raw.githubusercontent.com/almogtavor/yasa/docs-creation/docs/src/.vuepress/public/assets/logo.png",
```
Comment (kezhenxu94) [inline docs/src/.vuepress/config.js:39]:
```suggestion
      "https://raw.githubusercontent.com/yasa-org/yasa/docs-creation/docs/src/.vuepress/public/assets/logo.png",
```

File: docs/src/guide/contribution.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Contribution
+Find this project useful? 
+Any contribution is highly appreciated. You can open issues or pull requests to help make this project better.
+
+## Star YASA on Github!
+
+If you liked YASA we would be glad if you'd star us on [Github](https://github.com/yasa-org/yasa).
```
Comment (kezhenxu94) [inline docs/src/guide/contribution.md:7]:
```suggestion
If you liked YASA we would be glad if you'd star us on [GitHub](https://github.com/yasa-org/yasa).
```

**Review by kezhenxu94 — COMMENTED** [review]
Great work, @almogtavor when you think it's ready to deploy the docs in GitHub Pages please let me know and I can help to set it up, (or you want to help to set it up I can give the needed permissions

**Comment by almogtavor** [comment]
> Great work, @almogtavor when you think it's ready to deploy the docs in GitHub Pages please let me know and I can help to set it up, (or you want to help to set it up I can give the needed permissions

Thanks. I think the docs are now ready for initial deployment. If you can I'd be glad that you'll deploy (but permissions to do so will be great for future deployments). I think there is a need to explain more specifically about the features of YASA in comparison to Solr's old admin UI.

**Comment by almogtavor** [comment]
@kezhenxu94 Does the previous deployment is still available so we can share a link to that?

**Comment by kezhenxu94** [comment]
> @kezhenxu94 Does the previous deployment is still available so we can share a link to that?

I'm looking into that, it seems I had deployed gh-pages once but it's not available, maybe we can simply reuse that branch `gh-pages`, you should have write access to that branch

**Comment by kezhenxu94** [comment]
@almogtavor we might want to also set up GHA workflow to automatically build and push the doc to the branch

**Comment by kezhenxu94** [comment]
@almogtavor you can refer to https://github.com/yasa-org/yasa/blob/master/.github/workflows/DeployDemo.yaml to set up the workflow, and I'll create a new repo yasa-org.github.io to serve as the user github io website so that we can visit the doc via http://yasa-org.github.io

**Comment by kezhenxu94** [comment]
@almogtavor I've created and pushed an initial commit to https://github.com/yasa-org/yasa-org.github.io, serving at https://yasa-org.github.io , we can now set up a workflow to automatically deploy the doc website by referencing ☝🏻

**Comment by epugh** [comment]
Wow, that looks great.

**Comment by almogtavor** [comment]
> @almogtavor we might want to also set up GHA workflow to automatically build and push the doc to the branch

Agree

**Comment by almogtavor** [comment]
> > @kezhenxu94 Does the previous deployment is still available so we can share a link to that?
> 
> I'm looking into that, it seems I had deployed gh-pages once but it's not available, maybe we can simply reuse that branch `gh-pages`, you should have write access to that branch

Do you want the demo to be the Github page of YASA repo and that the documentation would be the GH page of the https://github.com/yasa-org/yasa-org.github.io repo?

**Comment by kezhenxu94** [comment]
> > > @kezhenxu94 Does the previous deployment is still available so we can share a link to that?
> 
> > 
> 
> > I'm looking into that, it seems I had deployed gh-pages once but it's not available, maybe we can simply reuse that branch `gh-pages`, you should have write access to that branch
> 
> 
> 
> Do you want the demo to be the Github page of YASA repo and that the documentation would be the GH page of the https://github.com/yasa-org/yasa-org.github.io repo?

Demo is in yasa-org/demo, it was indeed http://yasa-org.github.io but I had moved it to http://yasa-org.github.up/demo and now the doc site serves at http://yasa-org.guthub.io, i.e. the shortest url. We can consider using a custom domain name for it if needed

**Pull request merged**